docs/storage-layout-reorg-v0.10.md

```diff
@@ -0,0 +1,102 @@
+# Object Storage Layout Changes in v0.10
```

Comment:
Thanks for adding this!

I think it would make sense to update our docs to have an upgrades landing page that links to each target version so this stuff is easily findable in the future, but we should do that in another PR.

Reply:
Agree